src/parser/CodeBlock.h

```diff
@@ -282,36 +317,9 @@ class CodeBlock : public gc {
     CodeBlock() {}
     Context* m_context;
 
-    bool m_isConstructor : 1;
-    bool m_isStrict : 1;
-    bool m_hasCallNativeFunctionCode : 1;
-    bool m_isFunctionNameSaveOnHeap : 1;
-    bool m_isFunctionNameExplicitlyDeclared : 1;
-    bool m_canUseIndexedVariableStorage : 1;
-    bool m_canAllocateEnvironmentOnStack : 1;
-    bool m_needsComplexParameterCopy : 1;
-    bool m_hasEval : 1;
-    bool m_hasWith : 1;
-    bool m_hasSuper : 1;
-    bool m_hasCatch : 1;
-    bool m_hasYield : 1;
-    bool m_inCatch : 1;
-    bool m_inWith : 1;
-    bool m_usesArgumentsObject : 1;
-    bool m_isFunctionExpression : 1;
-    bool m_isFunctionDeclaration : 1;
-    bool m_isFunctionDeclarationWithSpecialBinding : 1;
-    bool m_isArrowFunctionExpression : 1;
-    bool m_isClassConstructor : 1;
-    bool m_isGenerator : 1;
-    bool m_isInWithScope : 1;
-    bool m_isEvalCodeInFunction : 1;
-    bool m_needsVirtualIDOperation : 1;
-    bool m_needToLoadThisValue : 1;
-    bool m_hasRestElement : 1;
-    uint16_t m_parameterCount;
-
+    ScopeInfo m_scopeInfo;
     AtomicString m_functionName;
+    uint16_t m_parameterCount;
```

Comment:
Could you reorder member variables?
ScopeInfo m_scopeInfo : <bits needed for ScopeInfo>;
uint16_t m_parameterCount;
AtomicString m_functionName;

reordering of member variables can reduce size of CodeBlock in 64bit system I think

Reply:
Of course, done :)